packages/lesswrong/components/common/RecommendedReading.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { Components, registerComponent } from 'meteor/vulcan:core';
+import { Link } from 'react-router';
+import withUser from '../common/withUser';
+import { withStyles } from '@material-ui/core/styles';
+import Hidden from '@material-ui/core/Hidden';
+
+const styles = theme => ({
+  smallScreenRecommendedReading: {
+    maxWidth: 720,
+    margin: "0 auto",
+  },
+});
+
+
+const RecommendedReading = ({currentUser, classes}) => {
+  const sectionTitle = currentUser ? "Recommended Sequences" : "Recommended Reading";
+  
+  const sectionWrapper = (recommendedContent) => (
+    <Components.Section
+      title={sectionTitle}
+      titleLink="/library"
+      titleComponent= {<Components.SectionSubtitle to="/library">
+        <Link to="/library">Sequence Library</Link>
+      </Components.SectionSubtitle>}
+    >{recommendedContent}</Components.Section>
+  );
+  
+  const omittedSectionWrapper = (content) => (
+    <div className={classes.smallScreenRecommendedReading}>
+      {content}
+    </div>
+  );
+  
+  if (currentUser) {
+    const suggestedSequences = count => (
+      <Components.SequencesGridWrapper
+        terms={{view:"curatedSequences", limit: count}}
+        showAuthor={true}
+        showLoadMore={false}
+        className={classes.frontpageSequencesGridList}
+      />
+    );
+    
+    return (<React.Fragment>
+      <Hidden smDown implementation="css">
+        {sectionWrapper(
+          suggestedSequences(3)
+        )}
+      </Hidden>
+      <Hidden mdUp xsDown implementation="css">
+        {omittedSectionWrapper(
+          suggestedSequences(3)
+        )}
+      </Hidden>
+      <Hidden smUp implementation="css">
+        {omittedSectionWrapper(
+          suggestedSequences(1)
+        )}
+      </Hidden>
+    </React.Fragment>);
+  } else {
+    return (<React.Fragment>
+      <Hidden smDown implementation="css">
+        {sectionWrapper(
```

Comment:
Why do this weird HoC thing instead of just directly writing the class here? I don't see any advantage of this over just adding a span here and just writing the className four times, which is about as much duplication as writing this function here four times, but makes the whole thing a lot more readable.

Reply:
If you want to keep it abstracted, since you expect to make future changes, then I would at least make it a normal component instead of an HoC, by using the `children` property.

Reply:
Huh? This avoids duplication of all the attributes of the `Components.Section`, which 6 lines long.

Reply:
Ah, sorry. I meant the smaller one below, which only abstracted away 2 lines.